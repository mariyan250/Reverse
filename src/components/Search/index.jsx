// React and Style
import React, { useState } from 'react';
import styles from './index.module.scss';

// Router
import { useHistory } from 'react-router-dom';

// Translation
import { useTranslation } from 'react-i18next';

function Search() {
  const [searchQuery, setSearchQuery] = useState('');

  const { t } = useTranslation();

  const history = useHistory();

  const handleSubmit = (e) => {
    e.preventDefault();
    history.push(searchQuery !== '' ? `/search?q=${searchQuery}` : '/');
  };

  return (
    <form className={`${styles.form} w-100 mx-auto`} aria-label="search" onSubmit={handleSubmit}>
      <div className="d-flex">
        <input
          type="text"
          placeholder={t('home.input.placeholder')}
          onChange={(e) => setSearchQuery(e.target.value)}
          value={searchQuery}
          aria-label="search"
          data-testid="searchInput"
        />

        <div>
          <button
            className={`${
              searchQuery ? 'app-bg-primary' : 'app-bg-secondary'
            } h-100 border-0 d-flex justify-content-center align-items-center`}
            aria-label="search"
          >
            <i className="fas fa-search app-text-white" />
          </button>
        </div>
      </div>
    </form>
  );
}

export default Search;
