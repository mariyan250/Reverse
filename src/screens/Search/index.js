// React
import React, { useEffect } from 'react';

// Connect Redux
import { connect } from 'react-redux';
import { getPosts } from '../../store/actions/postActions';

// Components
import Header from 'shared/components/Header';
import ItemList from './components/ItemList';

function Search(props) {
  useEffect(() => {
    props.getPosts();
  }, [props]);

  return (
    <main className="search">
      <Header />
      <ItemList items={props.posts} />
    </main>
  );
}

const mapStateToProps = (state) => ({ posts: state.posts.items });

export default connect(mapStateToProps, { getPosts })(Search);
