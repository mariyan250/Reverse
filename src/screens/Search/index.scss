// TO REM?

.search {
  & > h2 {
    margin-top: 7rem;
  }
}
