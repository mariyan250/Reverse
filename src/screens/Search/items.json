[
  {
    "id": 1,
    "name": "Lenovo",
    "category": "technology",
    "img": "https://www.lenovo.com/medias/lenovo-laptop-thinkpad-x1-carbon-gen8-subseries-hero.png?context=bWFzdGVyfHJvb3R8ODQ4NDd8aW1hZ2UvcG5nfGgyMi9oOGYvMTA2NzQ1ODc3Mjk5NTAucG5nfDQzODYxOTc5ODA0MWJhZTQyYThjOTAzZjE0NDI2NWVjYjY5MjE3MGFiMWEzODhlN2UyMGUwNGZhMWRmOTJmNzg",
    "price": "10$",
    "location": "Sofia",
    "seller": "Mariyan V."
  },
  {
    "id": 2,
    "name": "Carpet",
    "category": "home",
    "img": "https://pranekilimi.bg/upload/images/za-nas.jpg?1477653524940",
    "price": "10$",
    "location": "Sofia",
    "seller": "Mariyan V."
  },
  {
    "id": 3,
    "name": "Table",
    "category": "home",
    "img": "https://www.knoll.com/static_resources/images/products/catalog/eco/parts/642TR/642TR-KC_KC_FZ.jpg",
    "price": "10$",
    "location": "Plovdiv",
    "seller": "Mariyan V."
  },
  {
    "id": 4,
    "name": "Iphone X",
    "category": "technology",
    "img": "https://restore.bg/8407-thickbox_default/apple-iphone-x-64gb-space-gray.jpg",
    "price": "10$",
    "location": "Blagoevgrad",
    "seller": "Mariyan V."
  },
  {
    "id": 5,
    "name": "Lenovo",
    "category": "technology",
    "img": "https://www.lenovo.com/medias/lenovo-laptop-thinkpad-x1-carbon-gen8-subseries-hero.png?context=bWFzdGVyfHJvb3R8ODQ4NDd8aW1hZ2UvcG5nfGgyMi9oOGYvMTA2NzQ1ODc3Mjk5NTAucG5nfDQzODYxOTc5ODA0MWJhZTQyYThjOTAzZjE0NDI2NWVjYjY5MjE3MGFiMWEzODhlN2UyMGUwNGZhMWRmOTJmNzg",
    "price": "10$",
    "location": "Smolyan",
    "seller": "Mariyan V."
  },
  {
    "id": 6,
    "name": "Carpet",
    "category": "home",
    "location": "Rudozem",
    "img": "https://pranekilimi.bg/upload/images/za-nas.jpg?1477653524940",
    "price": "10$",
    "seller": "Mariyan V."
  },
  {
    "id": 7,
    "name": "Table",
    "category": "home",
    "location": "St.Vlas",
    "img": "https://www.knoll.com/static_resources/images/products/catalog/eco/parts/642TR/642TR-KC_KC_FZ.jpg",
    "price": "10$",
    "seller": "Mariyan V."
  },
  {
    "id": 4,
    "name": "Iphone X",
    "category": "technology",
    "img": "https://restore.bg/8407-thickbox_default/apple-iphone-x-64gb-space-gray.jpg",
    "price": "10$",
    "location": "Sliven",
    "seller": "Mariyan V."
  },
  {
    "id": 8,
    "name": "Lenovo",
    "location": "Ruse",
    "category": "technology",
    "img": "https://www.lenovo.com/medias/lenovo-laptop-thinkpad-x1-carbon-gen8-subseries-hero.png?context=bWFzdGVyfHJvb3R8ODQ4NDd8aW1hZ2UvcG5nfGgyMi9oOGYvMTA2NzQ1ODc3Mjk5NTAucG5nfDQzODYxOTc5ODA0MWJhZTQyYThjOTAzZjE0NDI2NWVjYjY5MjE3MGFiMWEzODhlN2UyMGUwNGZhMWRmOTJmNzg",
    "price": "10$",
    "seller": "Mariyan V."
  },
  {
    "id": 9,
    "name": "Carpet",
    "category": "home",
    "location": "Varna",
    "img": "https://pranekilimi.bg/upload/images/za-nas.jpg?1477653524940",
    "price": "10$",
    "seller": "Mariyan V."
  },
  {
    "id": 10,
    "name": "Table",
    "category": "home",
    "img": "https://www.knoll.com/static_resources/images/products/catalog/eco/parts/642TR/642TR-KC_KC_FZ.jpg",
    "price": "10$",
    "location": "Burgas",
    "seller": "Mariyan V."
  },
  {
    "id": 11,
    "name": "Iphone X",
    "category": "technology",
    "img": "https://restore.bg/8407-thickbox_default/apple-iphone-x-64gb-space-gray.jpg",
    "price": "10$",
    "location": "Pomorie",
    "seller": "Mariyan V."
  },
  {
    "id": 12,
    "name": "BMW M4",
    "category": "automobile",
    "img": "https://pvlimages.blob.core.windows.net/images/large/AG10769_20062019_24.jpg",
    "price": "120 000$",
    "location": "Pomorie",
    "seller": "Mariyan V."
  }
]
