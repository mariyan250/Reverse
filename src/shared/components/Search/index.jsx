// React and Style
import React, { useState } from 'react';
import './index.scss';

// Translation
import { useTranslation } from 'react-i18next';

function Search() {
  const [searchQuery, setSearchQuery] = useState('');
  const { t } = useTranslation();

  return (
    <form action="/search" className="search-form w-100 mx-auto">
      <article className="d-flex">
        <input
          type="text"
          name="query"
          placeholder={t('home_screen.input.placeholder')}
          onChange={(e) => setSearchQuery(e.target.value)}
          value={searchQuery}
          required
        />

        <article>
          <button
            className={`h-100 border-0 ${
              searchQuery ? 'app-bg-primary' : 'app-bg-secondary'
            } d-flex justify-content-center align-items-center`}
          >
            <i className="fas fa-search app-text-white" />
          </button>
        </article>
      </article>
    </form>
  );
}

export default Search;
