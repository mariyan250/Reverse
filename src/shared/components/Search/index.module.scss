@import '../../../index.scss';

.form {
  max-width: to-rem(500px);

  div {
    input {
      font-size: to-rem(16px);
      padding: to-rem(8px) to-rem(15px);

      @include media-breakpoint-md-up {
        padding: to-rem(8px) to-rem(15px);
      }
    }

    button {
      width: to-rem(52px);
      margin-left: to-rem(-3px);
      transition: $slow;

      @include media-breakpoint-md-up {
        padding: to-rem(12px) to-rem(14px);
        top: to-rem(-0.5px);
      }
    }

    i {
      font-size: to-rem(20px);
    }
  }
}
