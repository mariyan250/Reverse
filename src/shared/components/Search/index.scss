@import '../../../index.scss';

form {
  article {
    input {
      font-size: to-rem(16px);
      padding: to-rem(8px) to-rem(15px);

      @include media-breakpoint-md-up {
        padding: to-rem(8px) to-rem(15px);
      }
    }

    button {
      width: to-rem(52px);
      margin-left: to-rem(-3px);
      transition: 0.5s;

      @include media-breakpoint-md-up {
        padding: to-rem(12px) to-rem(14px);
        top: to-rem(-0.5px);
      }
    }

    i {
      font-size: to-rem(20px);
    }
  }
}
