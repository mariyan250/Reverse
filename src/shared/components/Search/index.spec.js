import React from 'react';
import { renderWithRouter } from 'tests/test-utils';
import Search from './index';

describe('Search', () => {
  let initialState;

  beforeAll(() => {
    initialState = {};
  });

  describe('Rendering', () => {
    it('should render the component with initial state', () => {
      const { container } = renderWithRouter(<Search />, initialState);

      expect(container.querySelector('.search-form')).toMatchSnapshot();
    });
  });
});
