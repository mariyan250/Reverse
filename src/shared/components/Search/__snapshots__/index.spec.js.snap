// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render the component 1`] = `
<div>
  <form
    action="/search"
    class="form w-100 mx-auto"
  >
    <article
      class="d-flex"
    >
      <input
        data-testid="searchInput"
        name="query"
        placeholder="home_screen.input.placeholder"
        required=""
        type="text"
        value=""
      />
      <article>
        <button
          class="h-100 border-0 app-bg-secondary d-flex justify-content-center align-items-center"
        >
          <i
            class="fas fa-search app-text-white"
          />
        </button>
      </article>
    </article>
  </form>
</div>
`;
